core/src/main/scala/kafka/log/Log.scala

```diff
@@ -2418,13 +2418,11 @@ class Log(@volatile private var _dir: File,
       info(s"Replacing overflowed segment $segment with split segments $newSegments")
       replaceSegments(newSegments.toList, List(segment))
       newSegments.toList
-    } catch {
-      case e: Exception =>
-        newSegments.foreach { splitSegment =>
-          splitSegment.close()
-          splitSegment.deleteIfExists()
-        }
-        throw e
+    } finally {
+      newSegments.foreach { splitSegment =>
+        splitSegment.close()
+        splitSegment.deleteIfExists()
+      }
```

Comment:
@dhruvilshah3 Is there a reason why we had a try/catch with a rethrow instead of using `finally`? spotBugs complained about it and hence the question.